pkg/smokescreen/smokescreen.go

```diff
@@ -552,9 +571,17 @@ func BuildProxy(config *Config) *goproxy.ProxyHttpServer {
 			return rejectResponse(pctx, pctx.Error)
 		}
 
-		// In case of an error, this function is called a second time to filter the
-		// response we generate so this logger will be called once.
-		logProxy(config, pctx)
+		if pctx.ConnectAction == goproxy.ConnectMitm {
+			// If the connection is a MITM
+			// 1 we don't want to log as it will be done in HandleConnectFunc
+			// 2 we want to close idle connections as they are not closed by default
+			// and CANONICAL-PROXY-CN-CLOSE is called on InstrumentedConn.Close
+			proxy.Tr.CloseIdleConnections()
```

Comment:
Are you just trying to force the log line here? Why not let the RoundTripper handle this?

Reply:
Also because the RoundTripper is shared I don't think there are any guarantees that you'll close the connection you're interested in. I think in most cases this will work as expected but is still racing against other inbound connections to the same domain.

Reply:
Yes this was for the log. In my previous test logs could take up to 30 s to appear. It's now immediate (on all manual tests stated in the description of this PR)
I've just retried and it doesn't happen anymore, I've modified some other part of the networking since these tests and opening the PR so this issue must have been fixed.

Moved `CloseIdleConnections` to the MITM test.